frontend/src/app/pages/category-edit/category-edit.component.ts

```diff
@@ -52,20 +52,27 @@ export class CategoryEditComponent implements OnInit {
 
   onSubmit() {
     const dialogRef = this.dialog.open(DialogWindowComponent);
-    dialogRef.afterClosed().subscribe((res) => {
-      switch (res.event) {
-        case "confirm-option":
-          this.categoryService.updateCategory(this.form.getRawValue()).subscribe(() => {
-            JSON.stringify(this.form.value);
-            this.router.navigate(['admin/categories'])
-          });
+    dialogRef.afterClosed().pipe(
```

Comment:
No need to unsubscribe from afterClosed() as it auto completes itself
https://stackoverflow.com/a/54923792